src/en/readcomicfree/src/eu/kanade/tachiyomi/extension/en/readcomicfree/ReadComicFree.kt

```diff
@@ -0,0 +1,261 @@
+package eu.kanade.tachiyomi.extension.en.readcomicfree
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ReadComicFree : ParsedHttpSource() {
+
+    override val name = "ReadComicFree"
+
+    override val baseUrl = "https://readcomicfree.org"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    override fun popularMangaSelector() = "div.items div.item"
+
+    override fun latestUpdatesSelector() = "nav > ul > li.row"
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/popular-comics".toHttpUrl().newBuilder().apply {
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/comic-updates".toHttpUrl().newBuilder().apply {
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = baseUrl.toHttpUrl().newBuilder()
```

Comment:
just use one `.apply` and add everything in there, this way you directly call `addQueryParameter`